biz.ganttproject.core/src/biz/ganttproject/core/chart/scene/gantt/TaskActivitySceneBuilder.java

```diff
@@ -190,38 +190,26 @@ private void processRegularActivity(int rowNum, A activity, OffsetList offsets,
       int rectHeight = rect.getHeight();
       int rectHalf = rectHeight / 2;
       int middleX = rect.getLeftX() + 3; // This is important to draw dependencies to/from milestones properly
-      Canvas.Polygon p = container.createPolygon(
-          middleX - rectHalf, rect.getMiddleY(),
-          middleX, rect.getMiddleY() - rectHalf,
-          middleX + rectHalf, rect.getMiddleY(),
-          middleX, rect.getMiddleY() + rectHalf);
-      p.setStyle("task.milestone");
+      Canvas.Rhombus r = container.createRhombus(
+          middleX - rectHalf, rect.getMiddleY() - rectHalf,
+          rectHeight, rectHeight
+      );
+      r.setStyle("task.milestone");
       //container.bind(p, activity);
-      resultShape = p;
+      resultShape = r;
     } else {
       Canvas.Rectangle nextRectangle = container.createRectangle(nextBounds.x, topy, nextLength, getRectangleHeight());
       resultShape = nextRectangle;
       if (nextHasNested || myTaskApi.isProjectTask(nextTask)) {
-        String prefix = myTaskApi.isProjectTask(nextTask) ? "task.projectTask" : "task.supertask";
-        nextRectangle.setStyle(prefix);
+        String style = myTaskApi.isProjectTask(nextTask) ? "task.projectTask" : "task.supertask";
+        nextRectangle.setStyle(style);
         if (myTaskApi.isFirst(activity)) {
-          Canvas.Polygon ending = container.createPolygon(nextRectangle.getLeftX(), nextRectangle.getTopY(),
-              nextRectangle.getLeftX() + nextRectangle.getHeight(), nextRectangle.getTopY(),
-              nextRectangle.getLeftX(), nextRectangle.getBottomY());
-          ending.setStyle(prefix + ".start");
-          ending.addStyle("task.ending");
-          myStyleApplier.applyStyle(ending);
-          polygons.add(ending);
+          nextRectangle.addStyle("task.container.open");
```

Comment:
please rename "container" to "summary". Summary task is a well-established term in this world. We also use "supertask" in our code, and that's also ok, but three terms is too much.

And ContainerRectangleRenderer to SummaryTaskRenderer